domain/infosync/label_manager.go

```diff
@@ -83,12 +83,17 @@ func (lm *PDLabelManager) GetLabelRules(ctx context.Context, ruleIDs []string) (
 	if err == nil && res != nil {
 		err = json.Unmarshal(res, &rules)
 	}
-	return rules, err
+
+	ruleMap := make(map[string]*label.Rule, len((rules)))
+	for _, r := range rules {
+		ruleMap[r.ID] = r
+	}
+	return ruleMap, err
 }
 
 type mockLabelManager struct {
 	sync.RWMutex
-	labelRules map[string]*label.Rule
+	labelRules map[string][]byte
```

Comment:
There  is no need to `marshal/unmarshak`, so you want a more real mock API?

Reply:
If we don't `marshal/unmarshal` it, we need to do type conversion explicitly.